AdminUi/apps/admin_ui/lib/pages/login_page.dart

```diff
@@ -17,6 +17,10 @@ class _LoginScreenState extends State<LoginScreen> {
   final _apiKeyController = TextEditingController();
   final _apiKeyFocusNode = FocusNode();
 
+  bool _isButtonEnabled = false;
+  bool apiKeyValid = false;
+  bool _attemptedLogin = false;
```

Comment:
Seems like a lot of variables for a simple login screen with a single text field. I don't want to know how this looks if there are 10-20 form fields. @jkoenig134 there must be a better way of doing this, right?